server.go

```diff
@@ -55,6 +56,12 @@ func NewServer(opts ...ServerOption) (*Server, error) {
 	if !c.Ephemeral {
 		// Apply migrations if file does not already exist
 		if _, err := os.Stat(c.DatabaseFilePath); os.IsNotExist(err) {
+			// Check if any of the parent dirs are missing
+			dir, _ := filepath.Split(c.DatabaseFilePath)
```

Comment:
I think there's a bug here when starting temporalite with a state file in the current directory, eg. `temporalite start -f foo.db`. From the godoc:

> If there is no Separator in path, Split returns an empty dir and file set to path.

In this case, `filepath.Split` returns `"", "foo.db"`, and calling `os.Stat("")` returns an error.

Instead we could use [filepath.Dir](https://pkg.go.dev/path/filepath#Dir) which returns `"."` when the path is empty.

```suggestion
			dir := filepath.Dir(c.DatabaseFilePath)
```

Reply:
I missed that part in the docs. Thank you.